libraries/abstractions/transport/secure_sockets/transport_secure_sockets.c

```diff
@@ -160,7 +160,7 @@ int32_t SecureSocketsTransport_Recv( NetworkContext_t * pNetworkContext,
                     pBuffer, bytesToRecv, ( void * ) pNetworkContext ) );
         bytesReceived = SOCKETS_EINVAL;
     }
-    else if( ( pNetworkContext != NULL ) && ( pNetworkContext->tcpSocket == SOCKETS_INVALID_SOCKET ) )
+    else if( pNetworkContext->tcpSocket == SOCKETS_INVALID_SOCKET )
```

Comment:
Looks like the 11.2 violations is due to `SOCKETS_INVALID_SOCKET`, which is `#define`d as `SOCKETS_INVALID_SOCKET    ( ( Socket_t ) ~0U )`, so this violation will appear every time the `SOCKETS_INVALID_SOCKET` macro is used. Since this macro is part of secure sockets, can allow this for now

Reply:
This happened due to the incomplete type from ```Socket_t ``` checked by 11.2 and seems like only happened in windows toolchain instead of linux toolchain. There may be something wrong in windows toolchain because this structure should be defined. So I think this one should be allowed.